### DISCUSSION TITLE
v2.9.0

### DISCUSSION THREAD

**Discussion by mehalter** [AstroNvim/AstroNvim#1428] [Announcements]
## What's Changed
- Add out of the box debugging support with `nvim-dap`, `nvim-dap-ui`, and `mason-nvim-dap` (Mac and Linux ONLY)
  - New Documentation: https://astronvim.github.io/Recipes/dap
  - New Mappings: https://astronvim.github.io/Basic%20Usage/mappings#debugger-mappings
- Automatically turn on/off `hlsearch`
  - can be disabled in `polish` function with `vim.on_key(nil, vim.api.nvim_get_namespaces()["auto_hlsearch"])`
- Briefly highlight yanked text with `vim.highlihgt.on_yank`
  - can be disabled in `polish` function with `vim.api.nvim_del_augroup_by_name("highlightyank")`
- Use Telescope for more LSP functions like go to definition/implementation/type definition/references/workspace symbols
- Various performance optimizations and code refactoring

**Full Changelog**: https://github.com/AstroNvim/AstroNvim/compare/v2.8.5...v2.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/AstroNvim/AstroNvim/releases/tag/v2.9.0'>v2.9.0</a>.</em>

**Comment by dohsimpson** [comment]
I used to have a custom setup for dap, but this works so much better! And I love the key mappings, very intuitive! Good work on this release!

**Reply by mehalter** [reply]
Thank you so much for the kind words!

**Comment by laurazard** [comment]
Just wanted to add, the changes to use Telescope for go to definition, implementation, etc. are a *huge* quality of life improvement. Thanks!! 🥳

**Comment by olejech** [comment]
> Use Telescope for more LSP functions like go to definition/implementation/type definition/references/workspace symbols

@mehalter 
When I use Telescope I got 2 results:
1. `*.d.ts` file from node_modules
2. target file (React component).

I fixed this issue [this filter](https://github.com/olejech/astronvim_config/blob/41b1ad322ba163a57fa2a81a4b9a69861cdc1a57/lsp/tsserver.lua#L11) when I use quicklist, but using Telescope I have the save issue :( How I can fix it?

<img width="1244" alt="Снимок экрана 2022-12-18 в 21 09 52" src="https://user-images.githubusercontent.com/64708593/208308223-5ba4e0cc-fa99-49a1-aa8d-ed8f7559788c.png">
